app/services/IdentityService.scala

```diff
@@ -23,7 +24,7 @@ class IdentityService(identityApiClient: IdentityApiClient) extends LazyLogging
       (response.json \ "user").asOpt[IdUser]
     }
 
-  def registerGuest(personalData: PersonalData): Future[Either[GuestUserNotCreated, String]] = {
+  def registerGuest(personalData: PersonalData): Future[Option[UserId]] = {
```

Comment:
Because there is only one left case here and the handling of this scenario (logging) is already done within the identity client. Options are also easier to reason about.